src/app/search/search.component.html

```diff
@@ -95,7 +95,8 @@
                     </mat-checkbox>
                   </span>
                 </div>
-                <span>{{ orderedBuckets.get(key).Items.get(subBucket) }}</span>
+                <span class="number-of-items">{{ orderedBuckets.get(key).Items.get(subBucket) }}</span>
+                <div class="histogram" [ngStyle]="{ 'width.px': getHistogramWidth(orderedBuckets.get(key).Items.get(subBucket)) }"></div>
```

Comment:
Can you please make sure that `getHistogramWidth` is not being invoked "zillions" of times? In general, you run that risk when you're referencing a method from the HTML template.

Reply:
Checked it, the function runs for every facet that gets rendered. For example upon the initial page load, only runs for the facets that appear in each accordion, and when you expand individual accordions the new rows that appear will then call the function